### PR TITLE
EWPP-223: Default exposed filters.

### PR DESCRIPTION
**Pull request by hernani**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: src/Plugin/EntityMetaRelation/ListPage.php [hunk 1]
```diff
@@ -355,14 +374,14 @@ protected function getBundleOptions(string $selected_entity_type): array {
    *   The current entity meta wrapper.
    *
    * @return array
```
Comment (imanoleguskiza) [inline src/Plugin/EntityMetaRelation/ListPage.php:376]: We need to update the return type here to array|null. Although I don't understand why we change the type here, we could keep returning an empty array and then when building the configuration check if is empty.

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]

File: config/schema/oe_list_pages.schema.yml [hunk 2]
```diff
@@ -21,6 +21,12 @@ node.type.*.third_party.oe_list_pages:
       label: 'The view mode to use when rendering the node in lists'
     default_sort:
       type: oe_list_pages.list_sort
+    default_exposed_filters:
```
Comment (upchuk) [inline config/schema/oe_list_pages.schema.yml:24]: Media don't need default exposed filters?

Also, seeing that the entire third party setting is the same across the two, we can extract the schema to its own thing no? Like I did with the sort.

File: config/schema/oe_list_pages.schema.yml [hunk 3]
```diff
@@ -21,6 +21,12 @@ node.type.*.third_party.oe_list_pages:
...
+      type: sequence
+      label: 'Exposed filters'
+      sequence:
+        type: string
+        label: 'Exposed filter'
```
Comment (upchuk) [inline config/schema/oe_list_pages.schema.yml:29]: What is this? The ID of the facet? Let's specify so we remember later.

File: src/Plugin/EntityMetaRelation/ListPage.php [hunk 4]
```diff
@@ -230,12 +230,25 @@ public function build(array $form, FormStateInterface $form_state, ContentEntity
     // Get available filters.
     if ($list_source && $available_filters = $list_source->getAvailableFilters()) {
       $configuration = $this->getExposedFiltersConfiguration($list_source, $entity_meta_wrapper);
+
+      // Override checkbox.
+      $form[$key]['bundle_wrapper']['exposed_filters_wrapper']['exposed_filters_override'] = [
+        '#type' => 'checkbox',
+        '#title' => $this->t('Override default exposed filters'),
```
Comment (upchuk) [inline src/Plugin/EntityMetaRelation/ListPage.php:237]: `Override default exposed filters`

Let's add also a description: `Configure which exposed filters should show up on this page.`

File: src/Plugin/EntityMetaRelation/ListPage.php [hunk 5]
```diff
@@ -230,12 +230,25 @@ public function build(array $form, FormStateInterface $form_state, ContentEntity
...
+        '#default_value' => isset($configuration),
```
Comment (upchuk) [inline src/Plugin/EntityMetaRelation/ListPage.php:238]: Let's not have `getExposedFiltersConfiguration()` return NILL and work out the override that means no filters somehow else. Maybe keep anther config option that says there is no override. And an empty array would mean none of the filters should be used. What do you think?

**Review by upchuk — CHANGES_REQUESTED** [review]
No tests?

**Review by imanoleguskiza — APPROVED** [review]

**Review by upchuk — APPROVED** [review]

**Pull request merged**